### PR TITLE
[DO-2201] added data correction method dor declaration in wrong format

### PR DESCRIPTION
**Pull request by EmmaSamoilova**

### CODE REVIEW

**Review by Andrii17s — APPROVED** [review]

File: business_register/converter/declaration.py [hunk 1]
```diff
@@ -213,6 +214,23 @@ def find_third_person_rights(self, right_data, object_name):
                 )
         return owner_type, full_name, company, company_name
 
+    def correction_data(self, declaration_data):
+        for step in declaration_data:
+            if step in ('step_0', 'step_1') or not declaration_data[step].get('data'):
+                continue
+            step_items = list(declaration_data[step]['data'].items())
+            for data in step_items:
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:222]:
```suggestion
            step_data = self.get_step_data(step, declaration_data)
            if step in ('step_0', 'step_1') or not step_data:
                continue
            for key, data in step_data.items():
```

File: business_register/converter/declaration.py [hunk 2]
```diff
@@ -213,6 +214,23 @@ def find_third_person_rights(self, right_data, object_name):
...
+                if step == 'step_2':
+                    data[1].update({'id': data[0]})
+                if data[1].get('rights'):
+                    rights = list(data[1]['rights'].items())
+                    for right in rights:
+                        if not right[1].get('rightBelongs') and right[0] in ('j', '1'):
+                            right[1].update({'rightBelongs': right[0]})
+                    data[1]['rights'] = [right[1] for right in rights]
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:230]:
```suggestion
                    data[1]['rights'] = rights.values()
```

File: business_register/converter/declaration.py [hunk 3]
```diff
@@ -213,6 +214,23 @@ def find_third_person_rights(self, right_data, object_name):
...
+            declaration_data[step]['data'] = [item[1] for item in step_items]
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:231]:
```suggestion
            declaration_data[step]['data'] = step_data.values()
```

File: business_register/converter/declaration.py [hunk 4]
```diff
@@ -213,6 +214,23 @@ def find_third_person_rights(self, right_data, object_name):
...
+            if step in ('step_0', 'step_1') or not declaration_data[step].get('data'):
+                continue
+            step_items = list(declaration_data[step]['data'].items())
+            for data in step_items:
+                if step == 'step_2':
+                    data[1].update({'id': data[0]})
+                if data[1].get('rights'):
+                    rights = list(data[1]['rights'].items())
+                    for right in rights:
+                        if not right[1].get('rightBelongs') and right[0] in ('j', '1'):
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:228]: replace `'j', '1'` with constants

File: business_register/converter/declaration.py [hunk 5]
```diff
@@ -213,6 +214,23 @@ def find_third_person_rights(self, right_data, object_name):
...
 
+    def correction_data(self, declaration_data):
+        for step in declaration_data:
+            if step in ('step_0', 'step_1') or not declaration_data[step].get('data'):
+                continue
+            step_items = list(declaration_data[step]['data'].items())
+            for data in step_items:
+                if step == 'step_2':
+                    data[1].update({'id': data[0]})
+                if data[1].get('rights'):
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:225]:
```suggestion
                if 'rights' in data:
```

File: business_register/converter/declaration.py [hunk 6]
```diff
@@ -213,6 +214,23 @@ def find_third_person_rights(self, right_data, object_name):
...
         return owner_type, full_name, company, company_name
 
+    def correction_data(self, declaration_data):
+        for step in declaration_data:
+            if step in ('step_0', 'step_1') or not declaration_data[step].get('data'):
+                continue
+            step_items = list(declaration_data[step]['data'].items())
+            for data in step_items:
+                if step == 'step_2':
+                    data[1].update({'id': data[0]})
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:224]:
```suggestion
                    data['id'] = key
```

File: business_register/converter/declaration.py [hunk 7]
```diff
@@ -213,6 +214,23 @@ def find_third_person_rights(self, right_data, object_name):
...
+                if data[1].get('rights'):
+                    rights = list(data[1]['rights'].items())
+                    for right in rights:
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:227]:
```suggestion
                    for key, right_data in data['rights'].items():
```

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/converter/declaration.py (hunk 1 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/converter/declaration.py (hunk 2 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/converter/declaration.py (hunk 3 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/converter/declaration.py (hunk 5 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/converter/declaration.py (hunk 6 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/converter/declaration.py (hunk 7 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

**Review by IegorSoboliev — APPROVED** [review]

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**